src/OpenLoco/CompanyManager.cpp

```diff
@@ -152,8 +152,13 @@ namespace OpenLoco::CompanyManager
             auto company = get(id);
             if (company != nullptr && !isPlayerCompany(id) && !company->empty())
             {
-                updatingCompanyId(id);
-                company->aiThink();
+                // Only the host should update AI, AI will run game commands
+                // which will be sent to all the clients
+                if (!isNetworked() || isNetworkHost())
+                {
+                    updatingCompanyId(id);
+                    company->aiThink();
+                }
```

Comment:
I wonder if this is the right way actually, since the game doesn't do that in the original I would assume that it handles AI specific game commands probably without having to send them since you would expect the AI to be deterministic.

Reply:
The original game's multiplayer was player vs player only. No AI, no more than two players.

Reply:
I see, I would still argue that the AI should be deterministic and not network commands that it uses during the simulation. In OpenRCT2 we also have some game actions being directly invoked when its used from the game logic during a tick.

Reply:
It isn't something I am worried about at the moment. Low priority.